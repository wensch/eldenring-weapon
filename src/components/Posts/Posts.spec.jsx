import { Posts } from '.';

const { render, screen } = require("@testing-library/react")

const props = {
  posts: [
    {
      id: '0',
      name: 'Name',
      description: 'text',
      image: 'image.png'
    },
    {
      id: '1',
      name: 'Name1',
      description: 'text1',
      image: 'image.png1'
    },
    {
      id: '2',
      name: 'Name2',
      description: 'text2',
      image: 'image.png2'
    }
  ]

}
describe('<Posts />', () => { 

  it('should render posts', () => {
    render(<Posts {...props} />)

    expect(screen.getAllByRole('heading', {name: /name/i})).toHaveLength(3)
    expect(screen.getAllByRole('img', {image: /png/i})).toHaveLength(3)
    expect(screen.getAllByText(/text/i)).toHaveLength(3)

  })
  
  it('should match snapshot', () => {
    const {container} = render(<Posts {...props} />)

    
    expect(container.firstChild).toMatchSnapshot()

  })
  
})