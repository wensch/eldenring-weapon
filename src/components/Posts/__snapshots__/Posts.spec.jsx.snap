// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Name"
      src="image.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Name
      </h2>
      <p>
        text
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Name1"
      src="image.png1"
    />
    <div
      class="post-content"
    >
      <h2>
        Name1
      </h2>
      <p>
        text1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Name2"
      src="image.png2"
    />
    <div
      class="post-content"
    >
      <h2>
        Name2
      </h2>
      <p>
        text2
      </p>
    </div>
  </div>
</div>
`;
